homeassistant/components/envisalink/switch.py

```diff
@@ -0,0 +1,87 @@
+"""Support for Envisalink zone bypass switches."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import (
+    CONF_ZONENAME,
+    DATA_EVL,
+    SIGNAL_ZONE_BYPASS_UPDATE,
+    ZONE_SCHEMA,
+    EnvisalinkDevice,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Envisalink switch entities."""
+    if not discovery_info:
+        return
+    configured_zones = discovery_info["zones"]
+
+    entities = []
+    for zone_num in configured_zones:
+        entity_config_data = ZONE_SCHEMA(configured_zones[zone_num])
+        zone_name = f"{entity_config_data[CONF_ZONENAME]}_bypass"
+        _LOGGER.debug("Setting up zone_bypass switch: %s", zone_name)
+
+        entity = EnvisalinkSwitch(
+            hass,
+            zone_num,
+            zone_name,
+            hass.data[DATA_EVL].alarm_state["zone"][zone_num],
+            hass.data[DATA_EVL],
+        )
+        entities.append(entity)
+
+    async_add_entities(entities)
+
+
+class EnvisalinkSwitch(EnvisalinkDevice, SwitchEntity):
+    """Representation of an Envisalink switch."""
+
+    def __init__(self, hass, zone_number, zone_name, info, controller):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.